### PR TITLE
show error on image upload

### PR DESCRIPTION
**Pull request by SaraVieira**
## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [x] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [x] conceptual
  - [x] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [x] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

closes https://github.com/webstudio-is/webstudio-designer/issues/37

### CODE REVIEW

File: apps/designer/app/routes/designer/$id.tsx [hunk 1]
```diff
@@ -88,11 +89,12 @@ export const action: ActionFunction = async ({ request, params }) => {
 const DesignerRoute = () => {
   const actionData = useActionData();
   const data = useLoaderData<Data | Error>();
+
   if ("errors" in data) {
     return <ErrorMessage message={data.errors} />;
   }
   if (actionData && "errors" in actionData) {
-    return <ErrorMessage message={actionData.errors} />;
+    toast.error(actionData.errors, { id: "upload-error" });
```
Comment (SaraVieira) [inline apps/designer/app/routes/designer/$id.tsx:97]: The id makes sure only one of them appears

File: packages/asset-uploader/src/upload.ts [hunk 2]
```diff
@@ -22,31 +21,42 @@ export const uploadAssets = async ({
...
+        projectId,
+        formData,
+      });
+    }
+  } catch (error) {
+    if ("maxBytes" in error) {
+      throw new Error(
+        `Asset cannot be bigger than ${commonUploadVars.MAX_UPLOAD_SIZE}MB`
+      );
+    }
```
Comment (SaraVieira) [inline packages/asset-uploader/src/upload.ts:57]: Getting a sad type error here :(

```
Object is of type 'unknown'.
```

Comment (kof) [reply packages/asset-uploader/src/upload.ts:57]: let me know if you need help, can check this out tomorrow

File: apps/designer/app/routes/designer/$id.tsx (hunk 1 above)
Comment (kof) [reply apps/designer/app/routes/designer/$id.tsx:97]: hmm, but if there are multiple once, shouldn't they all be somehow visible to the user? should they potentially get merged into one toast?

File: packages/asset-uploader/src/upload.ts (hunk 2 above)
Comment (SaraVieira) [reply packages/asset-uploader/src/upload.ts:57]: Yeah, lets check it out tomorrow

File: apps/designer/app/routes/designer/$id.tsx (hunk 1 above)
Comment (SaraVieira) [reply apps/designer/app/routes/designer/$id.tsx:97]: The issue is that it was running the same error twice

Comment (kof) [reply apps/designer/app/routes/designer/$id.tsx:97]: hmm, why though?

Comment (SaraVieira) [reply apps/designer/app/routes/designer/$id.tsx:97]: All very valid questions ahaha

It may be because it should be in a useEffect, let me check

Comment (kof) [reply apps/designer/app/routes/designer/$id.tsx:97]: 100%, you also don't need that in SSR and useEffect won't run there

Comment (SaraVieira) [reply apps/designer/app/routes/designer/$id.tsx]: Adding a useEffect did fix it

**Pull request merged**